src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -5510,7 +5510,8 @@ namespace VariableSpeedCoils {
             if (LatDegradModelSimFlag) {
                 // Calculate for SHReff using the Latent Degradation Model
                 if (NumIteration == 1) {
-                    QLatRated = QLoadTotal - QSensible;
+                    //match calculation in DXCoils at line 9899
+                    QLatRated = VarSpeedCoil(DXCoilNum).MSRatedTotCap(SpeedCal) * (1.0 - VarSpeedCoil(DXCoilNum).MSRatedSHR(SpeedCal));
```

Comment:
humm, I am not sure about this.  I defer to @rraustad

Reply:
@EnergyArchmage Here is what I found. The original calculation makes QLatRated = 0.0, so that no latent degradation occurs. I just copied the code from DXCoils. It seems working. The unit test shows what happens.

Reply:
These 2 equations should give the same answer. However, as @lgu1234 found, this model for some reason does set QLatRated = 0 (I just can't remember when or why that happened). The problem I see here is that if this equation is changed then where is QSensible corrected (or is it even needed on the fist pass)? This change only happens for NumIteration == 1 and should provide the correct QLatRated to use in the latent degradation model (but why didn't we see this in the Trane work?). I think I would rather see a correction to the total/sensible split during NumIteration == 1 instead of just correcting it after rated total and sensible capacity are calculated. That way this equation would not have to change and there would be confidence in the first pass of this model. I would also like to know if QLatRated is ever different during the simulation and if the need for 2 passes could be eliminated, but that's for a different issue. Maybe there are multiple QLatRated for each speed but I would think these would all be constants. This model will likely be deprecated (just don't know when) so maybe it's not worth the extra effort.

Reply:
@EnergyArchmage @rraustad What I changed is based on the definition of QLatRated. Ideally, the NumIteration should be removed. Suggested code is listed below: 

            if (LatDegradModelSimFlag) {
                    QLatRated = VarSpeedCoil(DXCoilNum).MSRatedTotCap(SpeedCal) * (1.0 - VarSpeedCoil(DXCoilNum).MSRatedSHR(SpeedCal));
                    QLatActual = QLoadTotal - QSensible;
                    SHRss = QSensible / QLoadTotal;
                    SHReff = CalcEffectiveSHR(
                        DXCoilNum, SHRss, CyclingScheme, RuntimeFrac, QLatRated, QLatActual, LoadSideInletDBTemp, LoadSideInletWBTemp);
                    //       Update sensible capacity based on effective SHR
                    QSensible = QLoadTotal * SHReff;
                    goto LOOP_exit;
                }
            } else {

Any comments?

Reply:
@EnergyArchmage @rraustad The removal of NumIteration is not correct. I retreat from my previous comments. I think to keep current fix and no more extra effort.

Reply:
@rraustad  I checked the code again in both variablespeed coil ad DXCoils. Here is a line 9899 in DXCoils:
                QLatRated = DXCoil(DXCoilNum).RatedTotCap(Mode) * (1.0 - DXCoil(DXCoilNum).RatedSHR(Mode)); // always the same number

QLatTated is always a constant based on inputs. 

Then I find that QLoadTotal is actual total capacity (with curve multipliers) via Debug and obtained by calling CalcTotCapSHR_VSWSHP. At the same time, SHR is also obtained from the same as a real SHR. Therefore, the difference between QLoadTotal - QSensible = QLoadTotal*(1-SHR) is a real latent capacity. The original code is a bug. I did a right correction.

Reply:
QLatRated should always be the same number, shouldn't it? It's the RATED latent capacity. This happens on the first loop in the calc routine. Then the load is calculated. Where QLoadTotal, Qsensible and Qlatent is found on the second loop. With your change you don't even need to do the first loop. Or is the first loop necessary for some other reason?

Reply:
@rraustad The change I made is to correct QLatRated calculation, so that the latent degradation is made to work. Otherwise, no SHR change at all. You may look at the unit test to find it out. 
I am not sure if I need to do the first loop or not, because there are other calculations available in the first loop. I will post a new issue to deal with loop structure to place the QLatRated calculation in the right location, since issue you raised is beyond the scope of this issue.

Reply:
I'm running the defect file now and see that the model is calculating QLoadTotal and SHR so that QLatRated = QLoadTotal - QSensible. So why then was this line of code needed? At the very least show a plot of before and after this line is changed.

Reply:
I just checked the Trane code where the latent degradation model was corrected. This line of code (line 5514) was not changed and the latent model does work. So it looks like this change is not needed. Plots showing total/sensible/latent capacity with and without this change would help determine if it's really needed.

Reply:
@rraustad I revised the program by adding two output variables:

Variable Speed Cooling Coil SHR
Variable Speed Cooling Coil Effective SHR

The first variable is assigned before calling latent degradation, while the second variable is assigned after calling latent degradation. Attached is a zipped file containing simulation results


old.csv shows simulation results using                 
if (NumIteration == 1) {
                    QLatRated = QLoadTotal - QSensible;

new.csv shows simulation results using                 
                if (NumIteration == 1) {
                    QLatRated = VarSpeedCoil(DXCoilNum).MSRatedTotCap(SpeedCal) * (1.0 - VarSpeedCoil(DXCoilNum).MSRatedSHR(SpeedCal));

 The third file is the difference summary file. Surprisely, the differences are very small.

 Then I debugged the program.


Using QLatRated = QLoadTotal - QSensible | Calculated |   |  
-- | -- | -- | --
  | Total | SHR | Sensible | Latent
  | 10089.88 | 0.807502 | 8147.598 | 1942.278

Using QLatRated = VarSpeedCoil(DXCoilNum).MSRatedTotCap(SpeedCal) * (1.0 - VarSpeedCoil(DXCoilNum).MSRatedSHR(SpeedCal));

-- | -- | -- | -- | --
  | Total | SHR | Sensible | Latent
  | 9246.065 | 0.79 | 7304.392 | 1941.674

The difference is small using two different calculations. I don't have preference to use which one. The code is back to the original QLatRated calculation.